src/gamma.jl

```diff
@@ -822,31 +799,38 @@ Compute the natural logarithm of the absolute value of the [`beta`](@ref) functi
 
 See also [`logbeta`](@ref).
 """
-function logabsbeta(a::Real, b::Real)
-    if a <= 0.0
-        if isinteger(a) && 1-a-b > 0
-            sgn = isinteger(b/2) ? 1 : -1
-            return logabsbeta(1-a-b,b)
-        end
+function logabsbeta(a::T, b::T) where T<:Real
+    # ensure that a <= b
+    if a > b
+        return logabsbeta(b, a)
     end
-    if b <= 0.0
-        if isinteger(b) && 1-a-b > 0
-            sgn = isinteger(a/2) ? 1 : -1
-            return logabsbeta(1-a-b,a)
+
+    if a <= 0.0 && isinteger(a)
+        if a + b <= 0
```

Comment:
This was
```suggestion
        if a + b <= 1
```
Is it changed on purpose?

Reply:
Yes, this is on purpose but it doesn't chance the result since `logbeta(0.0, b)` is also `Inf`. However, I realize that we also need to check `isinteger(b)` and add some tests.